io/net/net_tools.py

```diff
@@ -0,0 +1,359 @@
+            self.skip("alias for an loopback interface is configured")
+        self.alias = None
+        install_dependencies()
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+
+    @avocado.fail_on(process.CmdError)
+    def test_ifconfig(self):
+        """
+        Verify the functionality of the ifconfig
+        """
+        output = process.run("ifconfig", env={"LANG": "C"})
+        if "Local Loopback" not in output.stdout:
+            self.fail("unexpected output of ifconfig")
+        if self.ipv6:
+            if "inet6" not in output.stdout:
+                self.fail("Did not see IPV6 info")
+        # setup and verify an alias interface
+        self.alias = "lo:1"
+        process.system("ifconfig %s 127.0.0.240 netmask 255.0.0.0" %
+                       self.alias,
+                       sudo=True)
+        if "lo:1" not in process.system_output("ifconfig", env={"LANG": "C"}):
+            self.fail("Failed to configure alias for loopback")
+
+        self._remove_alias(self.alias)
+        self.alias = None
+        # Test to make loopback interface up and down
+        process.system("ifconfig lo down", sudo=True)
+        self.lo_up = False
+        if not process.system("ping -c 5 -w 5 localhost", ignore_status=True):
+            self.fail("Failed to change loopback interface to down state")
+
+    @staticmethod
+    def _remove_alias(name):
+        process.system("ifconfig %s down" % name,
+                       sudo=True)
+        if "lo:1" in process.system_output("ifconfig", env={"LANG": "C"}):
+            raise AssertionError("Failed to remove alias for loopback")
+
+    @staticmethod
+    def _restore_lo_intf_state():
+        process.system("ifconfig lo up", sudo=True)
+        if process.system("ping -c 5 -w 5 localhost", ignore_status=True):
+            raise AssertionError("Failed to restore loopback interface state")
+
+    def tearDown(self):
+        errs = []
+        if self.alias:
+            try:
+                self._remove_alias(self.alias)
+            except Exception:
+                errs.append("remove alias")
+        if not self.lo_up:
+            try:
+                self._restore_lo_intf_state()
+            except Exception:
+                errs.append("restore lo state")
+        if errs:
+            self.error("Failed to: %s" % ", ".join(errs))
+
+
+class Arp(Test):
+    """
+    arp: run it with different options. Check the return code.
+    """
+
+    def setUp(self):
+        interface_out = process.system_output("ip route show default",
+                                              env={"LANG": "C"})
+        if "default via" not in interface_out:
+            self.skip("No active interface with deafult gateway configured")
+        install_dependencies()
+        search_obj = re.search(
+            "^default via\s+(\S+)\s+dev\s+(\w+)", interface_out)
```

Comment:
This would look nicer:

```
        search_obj = re.search("^default via\s+(\S+)\s+dev\s+(\w+)",
                               interface_out)
```